### ISSUE TITLE
Number of bodies to scan not always right

### ISSUE THREAD

**Issue by JCalvertJoshua** [EDCD/EDDI#1332]
## EDDI version in which issue is found
EDDI 3.4.1-b2 (also in b1)

## VoiceAttack version in which issue is found (as applicable)
1.7.5

## Steps to reproduce
Use the discovery scanner ("Honk")

## Expected
Eddi telling the number of bodies to scan, or the number of already scanned bodies.

## Observed
The info Eddi gives is not always, I even think most of the times, the same as the FSS screen shows. 

## Investigation
In some systems EDDI says none of the bodies are scanned while the FSS screen shows 19/19

**Comment by Tkael** [comment]
Please provide logs and journal entries from the session where this occurred?

**Comment by JCalvertJoshua** [comment]
I'll make some tomorrow.

**Comment by JCalvertJoshua** [comment]
I did some testing and here are the logs and journals.
After some jumps I thought maybe the problem has to do with my adjusted EDDI-personality so I switched to the EDDI default personality about halfway.
Here are the log and journal of the first half (using my own personality):
[Journal.190601090856.01.log](https://github.com/EDCD/EDDI/files/3243626/Journal.190601090856.01.log)
[eddi1.log](https://github.com/EDCD/EDDI/files/3243628/eddi1.log)

And here the ones of the second half (EDDI default personality):
[Journal.190601093530.01.log](https://github.com/EDCD/EDDI/files/3243627/Journal.190601093530.01.log)
[eddi.log](https://github.com/EDCD/EDDI/files/3243630/eddi.log)

It doesn't look like the problem has to do with the personalities I used.
 
While making the jumps I opened up the FSS screen after EDDI told the number of bodies and checked it with the info on the screen to see if it was correct or not.
After the testrun I looked up the speech lines in my speechresponder.out file and combined all in this little spreadsheet and added comments about the jumps. 
[EDDI scanning progress.xlsx](https://github.com/EDCD/EDDI/files/3243640/EDDI.scanning.progress.xlsx)

I hope this is useful.

**Comment by Tkael** [comment]
Your journals only includes a handful of actual scan events (mostly auto-scans of bodies near jump exit points). It is the systems where those scans are located where the count comes out correct. Systems like "Sol", on the other hand, registered zero scan events in the player journal.

EDDI's count of scanned bodies can only include bodies you have personally scanned and that register scan events in the player journal. 
- New scan events are typically not written for bodies in a populated system, except when using a nav beacon scan to acquire complete system information.
- Scans completed by other cmdrs, even if those scans contribute to `progress`, likewise are not included. Explanation below.

`progress` (the only value which takes into account scans which have completed by other cmdrs) is not officially supported in the Discovery scan documentation because of a quirk in the nature of the reporting from FDev. 
1. Even though bodies can be discovered as a result of performing a discovery scan (like secondary stars), the Discovery scan is written before those other scans are registered. 
2. `progress` isn't calculated using secondary stars uncovered as a direct result of the discovery scan. This means that `progress` is often not accurate compared to what we see on-screen.
3. `progress` is weighted, so without knowledge of the complete system we cannot reverse engineer it to identify the total number of bodies scanned before the `Discovery scan` event or apply a correction for additional bodies identified as an immediate result of the `Discovery scan event`. 
We delay Discovery scan events slightly so that star scans generated as a result of a Discovery scan enter the queue before the Discovery scan event itself and are included correctly in our count, but we know that `progress` will often be incorrect.

TLDR: This is a feature for explorers. If you aren't actively scanning bodies and would prefer not to hear the scan count, you can disable the speech by either changing your ship's role to something other than `exploration` or `multi-purpose` or by revising the `System scan complete` script.

**Comment by JCalvertJoshua** [comment]
Thanks for the explanation. I thought I could use it to avoid having to open the FSS screen to check if there were bodies that weren't scanned yet after firing the discovery scanner when I enter a system but it seems that isn't possible. I'll make a command in my VA-profile to enable or disable the  speech and see what happens if i'm on my next exploration trip outside the bubble but from what you wrote I don't see why that will be any different.
